src/mlpack/methods/ann/vanilla_rbm_impl.hpp

```diff
@@ -76,45 +74,65 @@ void VanillaRBM<InitializationRuleType,
 
   if (!reset)
   {
-    ResetParameters();
+    Reset();
   }
 
   // Train the model.
   Timer::Start("rbm_optimization");
   const double out = optimizer.Optimize(parameter);
   Timer::Stop("rbm_optimization");
 }
-*/
-/* 
-* This function calculates
-* the free energy of the model
-*/
+
+
+ /* 
+  * This function calculates
+  * the free energy of the model
+  */
 template<typename InitializationRuleType,
          typename VisibleLayerType,
          typename HiddenLayerType>
-double VanillaRBM<InitializationRuleType, 
+double RBM<InitializationRuleType, 
            VisibleLayerType, 
            HiddenLayerType>::FreeEnergy(const arma::mat&& input)
 {
-  visible.FreeEnergy(input);
+  return visible.FreeEnergy(std::move(input));
+}
+
+template<typename InitializationRuleType,
+  typename VisibleLayerType,
+  typename HiddenLayerType>
+double RBM<InitializationRuleType, 
+  VisibleLayerType, 
+  HiddenLayerType>::Evaluate(const arma::mat& /* parameters*/, const size_t i)
+{
+  arma::mat freeEnergy, freeEnergyCorrupted, output;
```

Comment:
Ohhh sorry. I would change this. So there are 2 sources i am following
1. https://github.com/scikit-learn/scikit-learn/blob/14031f6/sklearn/neural_network/rbm.py#L28
2. http://deeplearning.net/tutorial/code/rbm.py you can have look at the pseudo likelihood cost function